Description: use new vmf_insert_pfn API to fix build with 4.20
 Since Linux 4.20-rc1 (commit ae2b01f370) vm_insert_pfn is gone. Use
 vmf_insert_pfn, which already returns the expected values.
Author: Luca Boccassi <bluca@debian.org>
--- a/nvidia-drm/nvidia-drm-gem-nvkms-memory.c
+++ b/nvidia-drm/nvidia-drm-gem-nvkms-memory.c
@@ -20,6 +20,8 @@
  * DEALINGS IN THE SOFTWARE.
  */
 
+#include <linux/version.h>
+
 #include "nvidia-drm-conftest.h"
 
 #if defined(NV_DRM_ATOMIC_MODESET_AVAILABLE)
@@ -237,6 +239,9 @@ static int __nv_drm_vma_fault(struct vm_
 
     page_offset = vmf->pgoff - drm_vma_node_start(&gem->vma_node);
 
+#if LINUX_VERSION_CODE >= KERNEL_VERSION(4, 20, 0)
+    return vmf_insert_pfn(vma, address, pfn + page_offset);
+#else
     ret = vm_insert_pfn(vma, address, pfn + page_offset);
 
     switch (ret) {
@@ -255,6 +260,7 @@ static int __nv_drm_vma_fault(struct vm_
     }
 
     return VM_FAULT_SIGBUS;
+#endif
 }
 
 /*
